### PR TITLE
Add FluentMultiLanguageLoader to allow easy translations access

### PR DESCRIPTION
**Pull request by saskenuba**
PR draft as discussed on https://github.com/kellpossible/cargo-i18n/issues/59

There is a lot of duplicated code because `FluentLanguageLoader` and `FluentMultiLanguageLoader` work the same way for loading and bundling, with exception of retrieval of translated strings. Perhaps moving some of this functionality as free functions?

Will work towards better docs of methods!

### CODE REVIEW

**Comment by kellpossible** [comment]
hey @saskenuba thanks again for the PR, sorry I haven't reviewed it yet, I'm in the middle of moving countries so I don't have much spare time, but hopefully in about a week I can get to it.

**Comment by saskenuba** [comment]
no problem @kellpossible! take a look when you have the time.

**Comment by saskenuba** [comment]
Hello @kellpossible! Sorry, it took so long to reply to this PR. I followed your thoughts and I believe the API is simplified now. Can you review it again?

Thanks!

**Comment by kellpossible** [comment]
hey @saskenuba thanks! I'll take another look at it as soon as I get the time :slightly_smiling_face:

File: i18n-embed/src/fluent/mod.rs [hunk 1]
```diff
@@ -113,7 +118,7 @@ impl FluentLanguageLoader {
     pub fn get_args_concrete<'source>(
         &self,
         message_id: &str,
-        args: HashMap<&'source str, FluentValue<'source>>,
+        args: HashMap<Cow<'source, str>, FluentValue<'source>>,
```
Comment (kellpossible) [inline i18n-embed/src/fluent/mod.rs:121]: This is a breaking change (see fl_macro failing unit test), was there are particular motivation for using `Cow` over `&str`?

**Review by kellpossible — COMMENTED** [review]
looks good! just the breaking change comment with `Cow` for the arguments map. Happy to accept the change if there's a good reason and we can update the failing unit test and treat this as a breaking update.

**Comment by kellpossible** [comment]
@saskenuba did you have a chance to look at this again?

**Comment by bikeshedder** [comment]
I'm in need for a working translation system that is able to switch the language between requests. Sometimes even within a request when generating multi-language responses. So this PR does exactly what I need. I'd like to jump in and complete this work if @kellpossible doesn't want to finish it.

I guess this PR needs to be rebased on the master branch and that breaking change (use of `Cow`) taken care of. Is there anything else that needs to be done for this to get merged?

**Comment by kellpossible** [comment]
@bikeshedder thanks for the offer of contribution, it is well received! That sounds like a good summary, there is also this https://github.com/kellpossible/cargo-i18n/pull/62#discussion_r596459712 that needs to be resolved.

File: i18n-embed/src/fluent/mod.rs [hunk 2]
```diff
@@ -373,3 +326,68 @@ impl LanguageLoader for FluentLanguageLoader {
         Ok(())
     }
 }
+
+/// Retrieve and load the file for the designated `LanguageIdentifier`
```
[the remark this answers is not in the document]
Comment (kellpossible) [reply i18n-embed/src/fluent/mod.rs]: I can see the unused variable has been removed but I haven't spent the time to see  how the error logic has been altered as a result of this.

**Comment by bikeshedder** [comment]
This PR can be closed now that #84 is merged.

**Pull request closed without merging**